shop/rest/renderers.py

```diff
@@ -3,6 +3,46 @@
 
 from rest_framework import renderers
 from rest_framework.compat import template_render
+from rest_framework.exceptions import APIException
+
+from shop.models.cart import CartModel
+from shop.serializers.cart import CartSerializer
+
+
+class ShopTemplateHTMLRenderer(renderers.TemplateHTMLRenderer):
+    """
+    Modified TemplateHTMLRenderer, which shall be used to render templates used by django-SHOP.
+    Instead of polluting the template context with the serialized data, that information is
+    stored inside a separate `data` attribute, which allows to add a Cart and Paginator object.
+
```

Comment:
It's not only about name collisions.
Django-CMS requires a few context variables on the global scope level, for instance the ``current_page``, the ``cms_menu_renderer``, the ``request`` object and many more. All these objects are pure Python instances. On the other side, data provided by the serializers are always dicts or lists of number or string primitives.
Therefore mixing these two sets of information would not only be dangerous, but inherently confusing. Therefore in my opinion having a dedicated dictionary of serialized data, is a much cleaner and safer solution.

Moreover, templates used for a mixed representation of CMS content and content emerging from REST serializers, *must* be written for the CMS specifically. Such templates normally contain tags such as ``{% render_menu %}``, ``{% cms_toolbar %}``, ``{% render_placeholder %}``, etc.

Therefore, these kind of templates will *always* be incompatible with templates used to render pure serialized data.